lib/orderbook/OrderBook.ts

```diff
@@ -98,7 +98,14 @@ class OrderBook extends EventEmitter {
           this.emit('ownOrder.swapped', { pairId, quantity, id: orderId });
         }
       });
-      // TODO: bind to other swap events
+      this.swaps.on('swap.failed', (deal) => {
+        if (deal.role === SwapRole.Maker && deal.phase === SwapPhase.SwapAgreed) {
+          // if our order is the maker and the swap failed after it was agreed to but before it was executed
+          // we must release the hold on the order that we set when we agreed to the deal
+          const tp = this.tradingPairs.get(deal.pairId);
```

Comment:
can use `this.getTradingPair` as well, since you'll get an error anyway if it's missing (but less informative one)